### DISCUSSION TITLE
what does warm-up instruction mean?

### DISCUSSION THREAD

**Discussion by JimmyWong1998** [ChampSim/ChampSim#377] [Q&A]
Hi ChampSim experts,

As a newbie in using ChampSim, I am confused on the meaning of the warm-up instruction in the command. Would different warm-up configurations impact the simulation result? And if I want to capture the run-time memory address information, should I include the address information in the warm-up stage?

Best Wish,
Jimmy

**Comment by ngober** [comment]
The warmup phase is used to fill the caches before the detailed simulation phase. It is expedited (latencies are ignored) and statistics from the warmup phase are discarded.

**Reply by JimmyWong1998** [reply]
Many thanks. I have one more question about this. When I put different trace files created by my own programms to simulate and then extract the memory address(e.g. access address of CPU packet and with the type LOAD and WRITE), I find there is always a set of similar memory address accessed at the begining, which seems something like bootloader phase. Is this phase just warmup phase? Furthermore, how can I extract the address trace that really corresponding to my own programm, not inlcuding the "bootloader" part?

**Reply by ngober** [reply]
The concept of "warmup" is a simulator concept, not a program concept. You could, if you choose, consider this initialization phase to be warmup. I think the best way to determine the beginning of your program would be to run it with a debugger and find the location of `main()`, then find the position of those instructions in the trace.
